site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -6251,37 +6251,29 @@ private List getLtiToolGroup(String groupName, File moreInfoDir, Site site) {
 		if (allTools != null && !allTools.isEmpty()) {
 			for (Map<String, Object> tool : allTools) {
 				Set keySet = tool.keySet();
-				String toolIdString = tool.get(m_ltiService.LTI_ID).toString();
-				boolean toolStealthed = tool.get(m_ltiService.LTI_VISIBLE).toString().equals("1");
-				boolean ltiToolSelected = ltiSelectedTools.contains(toolIdString); 
-
-				try
-				{
-					// in Oracle, the lti tool id is returned as BigDecimal, which cannot be casted into Integer directly
-					Integer ltiId = Integer.valueOf(toolIdString);
-					if (ltiId != null) {
-						String ltiToolId = ltiId.toString(); 
-						if (ltiToolId != null && (!toolStealthed || ltiToolSelected) ) {
-							String relativeWebPath = null;
-							MyTool newTool = new MyTool();
-							newTool.title = tool.get("title").toString();
-							newTool.id = LTITOOL_ID_PREFIX + ltiToolId;
-							newTool.description = (String) tool.get("description");
-							newTool.group = groupName;
-							relativeWebPath = getMoreInfoUrl(moreInfoDir, ltiToolId);
-							if (relativeWebPath != null) {
-								newTool.moreInfo = relativeWebPath;
-							}
-							// SAK16600 should this be a property or specified in  toolOrder.xml?
-							newTool.required = false; 
-							newTool.selected = ltiToolSelected;
-							ltiTools.add(newTool);
-						}
-					}
-				}
-				catch (NumberFormatException e)
-				{
-					M_log.error(this + " Cannot cast tool id String " + toolIdString + " into integer value.");
+				Long ltiVisible = FoormUtil.getLong(tool.get(m_ltiService.LTI_VISIBLE));
+				Integer ltiId = FoormUtil.getInt(tool.get(m_ltiService.LTI_ID));
+
```

Comment:
Both of these can be done via the more standard NumberUtils.toInt or NumberUtils.toLong with the arg as a string rather than another custom library.